src/client.cpp

```diff
@@ -421,7 +421,7 @@ namespace konstructs {
                                 if(received.find(pos) == received.end()) {
                                     if(requested.find(pos) == requested.end()) {
                                         /* We don't have it and didn't request it */
-                                        if(distance < best_chunk_distance) {
+                                        if(distance < best_chunk_distance && distance < r) {
```

Comment:
The corners are outside the view distance, so we will ask for chunks and drop them all the time.